packages/@uppy/companion/src/server/provider/zoom/index.js

```diff
@@ -0,0 +1,274 @@
+const Provider = require('../Provider')
+
+const request = require('request')
+const moment = require('moment')
+const purest = require('purest')({ request })
+const logger = require('../../logger')
+const adapter = require('./adapter')
+const { ProviderApiError, ProviderAuthError } = require('../error')
+
+const BASE_URL = 'https://zoom.us/v2'
+const GET_LIST_PATH = '/users/me/recordings'
+const GET_USER_PATH = '/users/me'
+const PAGE_SIZE = 300
+const DEFAULT_RANGE_MOS = 23
+
+/**
+ * Adapter for API https://marketplace.zoom.us/docs/api-reference/zoom-api
+ */
+class Zoom extends Provider {
+  constructor (options) {
+    super(options)
+    this.authProvider = options.provider = Zoom.authProvider
+    this.client = purest(options)
+  }
+
+  static get authProvider () {
+    return 'zoom'
+  }
+
+  list (options, done) {
+    /*
+    - returns list of months by default
+    - drill down for specific files in each month
+    */
+    const token = options.token
+    const query = options.query || {}
+    const { cursor, from, to } = query
+    const meetingId = options.directory || ''
+
+    if (!from && !to && !meetingId) {
+      const end = cursor && moment(cursor)
+      this.client.get(`${BASE_URL}${GET_USER_PATH}`)
+        .auth(token)
+        .request((err, resp, body) => {
+          if (err || resp.statusCode !== 200 || !body) {
+            return this._listError(err, resp, done)
+          }
+          done(null, this._initializeData(body, end))
+        })
+    }
+
+    if (from && to) {
+      const queryObj = {
+        page_size: PAGE_SIZE,
+        from,
+        to
+      }
+
+      if (cursor) {
+        queryObj.next_page_token = cursor
+      }
+
+      this.client.get(`${BASE_URL}${GET_LIST_PATH}`)
+        .qs(queryObj)
+        .auth(token)
+        .request((err, resp, body) => {
+          if (
+            err ||
+            resp.statusCode !== 200 ||
+            !body
```

Comment:
in what case is `body` falsy? 🤔

Reply:
I was getting a 200 and empty body when I was testing in postman earlier, I think it was around things i was expecting a 404 response on.  I can't seem to reproduce now so I'll remove it